library/test/src/test_result.rs

```diff
@@ -45,6 +45,18 @@ pub(crate) fn calc_result<'a>(
     time_opts: Option<&time::TestTimeOptions>,
     exec_time: Option<&time::TestExecTime>,
 ) -> TestResult {
+    let span_str = if desc.source_file != "" {
+        format!(" at {}:{}:{}", desc.source_file, desc.start_line, desc.start_col)
+    } else {
+        "".to_string()
+    };
+
+    let span_should_panic_with_message_didnot_panic = if desc.source_file != "" {
```

Comment:
Nit: Given this is a local variable, I'd generally choose a bit of a shorter name. It's also only applicable to one of the match arms, so move it within that block below (at line 63)